src/main/java/io/confluent/copycat/jdbc/DataConverter.java

```diff
@@ -225,10 +222,10 @@ private static void addFieldSchema(ResultSetMetaData metadata, int col,
       case Types.TIMESTAMP: {
         // Current approach uses UNIX time in milliseconds. May lose nanosecond precision
         // available in source type
-        if (nullableBuilder != null) {
-          nullableBuilder.longType().endUnion().noDefault();
+        if (optional) {
```

Comment:
We'll want to change this to a logical type once we have those in Copycat's API. For the time being, we might want to annotate specialized types like these at least with a name.